lib/hanami/cli/commands/app/db/migrate.rb

```diff
@@ -24,21 +30,46 @@ def call(target: nil, app: false, slice: nil, dump: true, command_exit: method(:
             private
 
             def migrate_database(database, target:)
-              return true unless migrations?(database)
-
               measure "database #{database.name} migrated" do
                 if target
                   database.run_migrations(target: Integer(target))
                 else
                   database.run_migrations
                 end
-
-                true
               end
             end
 
-            def migrations?(database)
-              database.migrations_dir? && database.sequel_migrator.files.any?
+            def migrations_dir_missing?(database)
+              !database.migrations_dir?
+            end
+
+            def no_migrations?(database)
+              database.sequel_migrator.files.empty?
+            end
+
+            def warn_on_missing_migrations_dir(database)
+              out.puts <<~STR
+                WARNING: Database #{database.name} expects migrations to be located within #{relative_migrations_dir(database)} but that folder does not exist.
+
+                No database migrations can be run for this database.
+              STR
+            end
+
+            def warn_on_empty_migrations_dir(database)
+              out.puts <<~STR
+                WARNING: Database #{database.name} has the correct migrations folder #{relative_migrations_dir(database)} but that folder is empty.
```

Comment:
I'm been thinking about this case. An empty migrations folder IMO doesn't necessarily indicate something has gone wrong.

Such an empty folder can be expected:

- In brand new apps
- And in apps where a user has collapsed historical migrations by ensuring the schema dump is up to date and deleting the old migration files.

In these cases, nothing is really _wrong_ (as opposed to the entire migrations dir being missing, which _is_ wrong), it's just that we don't have any migrations to apply, and I think we could consider that routine.

So, a couple options here:

1. Remove this warning, or
2. Tone down the language so it's more an "informative" message than a "WARNING" message

What do you think?

Another thing I'd like to consider is being smarter with our output when there _are_ migration files present but we're already up to date. Right now we just print the same output each time:

```
=> database db/my_app.sqlite migrated in 0.0002s
```

In this case, this is kind of a lie: the migrations are already up to date and we haven't actually "migrated" anything.

So perhaps we have the chance to unify the output across both the the "files absent" and "files present, but already up to date" cases.

If I could suggest something here, it'd be to remove this extra warning for now, and then give ourselves a ticket to consider making it nicer/more unified as a follow-up effort (and I'd probably mark it slightly lower priority than everything else we already have queued up before the 2.2 release).

Reply:
Gotcha! Yea it's definitely not a problem to have an empty migrations folder, I mostly just wanted the output to reflect that, especially to help people who might have the migrations in the wrong place (we generate them correctly, but maybe someone will manually want to add migrations to a slice after the fact).

I softened the WARNING to a NOTE. We can clean that up once we come back to it, to fix the issue you raised.

Do you think a comment on #177 is an appropriate place for that better output, or a new issue?

Reply:
@cllns Let's make a separate issue for this one. #177 is mostly about reviewing everything that already exists, rather than adding a specific new thing.

Reply:
I created #225 for this :)